### ISSUE TITLE
Feature request: manipulate sysctl variables (and do other things), inside NetNS

### ISSUE THREAD

**Issue by crosser** [svinota/pyroute2#697]
My concrete task is to set interface-specific sysctl variable (`arp_ignore`) for an interface inside netns. It is as easy as opening a `/proc/sys/net/.../arp_ignore` file and writing a character there. But it has to be done by a process running inside the namespace.

I thought that it would be useful to have a "lighter" version of `NSPopen`: an interface to run a chunk of pyrhon code inside NetNS. Ideally, it could look like this:
```
with NetNS("mynamespace"):
    with open("/proc/sys/net/ipv4/conf/{}/arp_ignore".format(ifname), "w") as f:
        f.write(str(arplevel))
```
Or maybe there is a better way?..

**Comment by ffourcot** [comment]
I think that we don't have any manager yet (that could be few lines of contextmanager with existing code). But you could have a look here :

https://github.com/svinota/pyroute2/blob/master/pyroute2/netns/__init__.py#L282

pushns() popns() should do the job for you sysctl need.

**Comment by crosser** [comment]
Indeed it works for me! Thanks a lot @ffourcot !

**Comment by svinota** [comment]
There is a little issue with context managers. Let's look at a hypothetical example:
```python
import threading


def worker():
    with netns('test'):
        with open('/proc/sys/net/…') as f:
            f.write('…')

th = threading.Thread(target=worker)
th.start()
do_whatever()
th.join()
```

Ok now we have a context manager in a thread. The issue is that between `__enter__()` and `__exit__()` **all** the operations in **all** the threads will be executed within a netns, **not only** those under the context manager call. Including `do_whatever()` — maybe. Or maybe not. Same will be true for async code.

If we find a way to transparently run the context manager code block in a separate process, then we may use it. Otherwise I see no way, the syntax will be pretty misleading, users will believe that only the context manager will be run in a netns.

**Comment by crosser** [comment]
I agree, but isn't the current situation even worse? Different threads pushing and popping uncoordinated against a single(?) global array?

In a context manager, you might at least use a mutex, which is less then ideal, but possibly better than total chaos?..

**Comment by svinota** [comment]
Not much better, I have to agree. But at least these ops are global, while a context manager creates 
a local context and semantically this context implies an isolated code flow. Changing netns is always a global operation that affects the whole process.

A mutex can help only if users will use it. If they avoid using it — then it's out of our control. We're changing netns for the whole process, and unrelated program flows will continue to work, probably relying on the environment — that has ben changed.

I'm not rejecting completely the idea of a context manager. I'm just trying to find a solution, when the context, created by the manager, will be truly local and will not affect all other possible program flows, be it async or threading.

A possible solution could be fork on `__enter__()` and then fetch `locals()`/`globals()` from the child on `__exit__()`. But it sounds a bit fragile, and not all the data can be pickled to be transferred between processes.

Still in doubts.

**Comment by crosser** [comment]
Thinking out loud:
Perhaps make an interface akin to `NSPopen` rather than a context manager. Semantics could be: take a callable object as an argument and evaluate it in a separate process. If the callable returns a serializable object, pass over this object to the caller. If the callable throws an exception or returns a non-serializable object, return an exception object (or just throw that exception).
